### PR TITLE
engine: upgrade manifest to spec

### PR DESCRIPTION
**Pull request by SparkYuan**
- With pull requests:
  - Open your pull request against `main`
  - Your pull request should have no more than two commits, if not you should squash them.
  - It should pass all tests in the available continuous integration systems such as GitHub Actions.
  - You should add/modify tests to cover your proposed code changes.
  - If your pull request contains a new feature, please document it on the README.

### CODE REVIEW

File: pkg/engine/models/spec.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package models
+
+// Spec represent the KCL compile result
+type Spec struct {
```
Comment (howieyuen) [inline pkg/engine/models/spec.go:4]: what's the abstraction of Spec for ? I don't see the necessity

File: pkg/engine/models/resource.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+type Resources []Resource
+
+type Resource struct {
+	// ID is the unique key of this resource in the whole State. ApiVersion:Kind:Namespace:Name is an idiomatic way of Kubernetes resources.
+	ID string `json:"id"`
+
+	// Type represents all Runtimes we supported
+	Type Type `json:"type"`
+
+	//Attributes represents all specified attributes of this resource
```
Comment (howieyuen) [inline pkg/engine/models/resource.go:14]: go lint issue

File: pkg/engine/models/spec.go (hunk 1 above)
Comment (SparkYuan) [reply pkg/engine/models/spec.go:4]: Spec means all desired states of resources in a stack

File: pkg/engine/operation/change.go [hunk 3]
```diff
@@ -5,10 +5,11 @@ import (
 	"fmt"
 	"strings"
 
+	"kusionstack.io/kusion/pkg/engine/models"
```
Comment (howieyuen) [inline pkg/engine/operation/change.go:8]: sort the import statements as gofmt suggest, same as below

**Review by howieyuen — CHANGES_REQUESTED** [review]

**Review by elliotxx — APPROVED** [review]
lgtm

**Review by howieyuen — APPROVED** [review]
lgtm

**Pull request merged**